packages/openapi-typescript/src/transform/schema-object.ts

```diff
@@ -126,8 +121,14 @@ export function defaultSchemaObjectTransform(schemaObject: SchemaObject | Refere
         itemType = tsArrayOf(itemType);
       }
       itemType = ctx.immutableTypes || schemaObject.readOnly ? tsReadonly(itemType) : itemType;
+
       return schemaObject.nullable ? tsUnionOf(itemType, "null") : itemType;
     }
+
+    // polymorphic, or 3.1 nullable
+    if (Array.isArray(schemaObject.type)) {
+      return tsUnionOf(...schemaObject.type.map((t) => transformSchemaObject({ ...schemaObject, type: t }, { path, ctx })));
```

Comment:
Big change: `tsUnionOf` rather than `tsOneOf` here is more correct, and now handles 3.1 nullable types correctly. There shouldn’t be any performance impact from this.